src/libYARP_math/src/math.cpp

```diff
@@ -232,6 +233,14 @@ Vector& yarp::math::operator*=(Vector &a, const Vector &b)
     return a;
 }
 
+Quaternion yarp::math::operator*(const Quaternion& a, const Quaternion& b)
+{
+    return Quaternion(a.w()*b.x() + a.x()*b.w() + a.y()*b.z() - a.z()*b.y(),
```

Comment:
According to [1, p. 101] this formula is correct (though the reference has a quaternion composed of real part-vector, and here we have a quaternion vector-real part). Probably, writing the formula in a compact, vectorial way would simplify the equation checking. 

[1]: https://tel.archives-ouvertes.fr/tel-00460801v1/document